monkestation/code/datums/elements/shoesteps.dm

```diff
@@ -0,0 +1,328 @@
+///A simple element that lets shoes have togglable custom sounds
+/datum/element/shoesteps
+	var/list/custom_sounds = list()
+	var/sounds = TRUE
+
+/datum/element/shoesteps/Attach(atom/movable/target)
+	. = ..()
+	RegisterSignal(target, COMSIG_ATOM_EXAMINE, PROC_REF(on_examine))
+	RegisterSignal(target, COMSIG_CLICK_CTRL, PROC_REF(on_ctrl_click))
+	RegisterSignal(target, COMSIG_ITEM_EQUIPPED, PROC_REF(on_equip))
+	RegisterSignal(target, COMSIG_ITEM_DROPPED, PROC_REF(on_drop))
+
+/datum/element/shoesteps/Detach(atom/source)
+	. = ..()
+	UnregisterSignal(source, COMSIG_ATOM_EXAMINE)
+	UnregisterSignal(source, COMSIG_CLICK_CTRL)
+	UnregisterSignal(source, COMSIG_ITEM_EQUIPPED)
+	UnregisterSignal(source, COMSIG_ITEM_DROPPED)
+
+/datum/element/shoesteps/proc/on_examine(datum/source, mob/user, list/examine_text)
+	SIGNAL_HANDLER
+
+	examine_text += span_notice("There seems to be a noisemaker inside, which will change your walking sounds. You can enable or disable it using Ctrl Click.")
+	examine_text += span_notice("The noisemaker is currently [(sounds) ? "on" : "off"].")
+
+/datum/element/shoesteps/proc/on_ctrl_click(datum/source, mob/living/carbon/clicker)
+	SIGNAL_HANDLER
+
+	INVOKE_ASYNC(src, PROC_REF(toggle_sounds), clicker)
+
+/datum/element/shoesteps/proc/toggle_sounds(mob/living/carbon/clicker)
+	if(do_after(clicker, 1.5 SECONDS))
+		sounds ^= 1
+		to_chat(clicker, span_warning("[(sounds) ? "You turn on the noisemaker." : "You turn the noisemaker off."]"))
```

Comment:
```suggestion
		sounds = !sounds
		to_chat(clicker, span_warning("You turn the noisemaker [sounds ? "on" : "off"]."))
```

also, it'd be nice if you defined like a `DOAFTER_SOURCE_TOGGLE_SHOESTEPS` and used that as the `interaction_key` for `do_after`